components/ModalRoot.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+  <SfModal class="sf-modal" :visible="!!componentRef" @close="handleClose">
+    <template #modal-bar>
+      <SfBar class="sf-modal__bar bar-heading" :title="titleRef" @click:close="handleClose" />
+    </template>
+    <component :is="componentRef" :properties="properties" @onClose="handleClose" />
+  </SfModal>
+</template>
+<script lang="ts">
+import { SfModal, SfSearchBar, SfBar } from "@storefront-ui/vue"
+import { ref } from "@vue/composition-api"
+import { ModalBus } from "../eventBus/eventBus"
+export default {
+  components: { SfModal, SfSearchBar, SfBar },
+  setup() {
+    const componentRef = ref(null)
+    const titleRef = ref("")
+    const properties = ref(null)
+
+    const handleClose = () => {
+      componentRef.value = null
+    }
+
+    ModalBus.$on("open", ({ component, props }) => {
+      componentRef.value = component
+      titleRef.value = "Select Store"
```

Comment:
"Select Store" should not be hardcoded for the title

Reply:
Created a function that will generate the title based on Modal Component name.